truffle/src/main/java/org/jruby/truffle/runtime/object/ObjectGraph.java

```diff
@@ -115,9 +129,23 @@ private void visitObject(DynamicObject object, ObjectGraphVisitor visitor) throw
 
             // Visit all properties
 
-            for (Property property : object.getShape().getProperties()) {
+            for (Property property : object.getShape().getPropertyListInternal(false)) {
```

Comment:
We want to visit all fields, whether they're instance variables or they're internal fields (like `store` in `Array`) to find all objects. Things like `store` are currently `HiddenKey` so that they aren't mistaken for instance variables.